snakesist/exist_client.py

```diff
@@ -165,35 +178,50 @@ class ExistClient:
     :param user: username
     :param password: password
     :param prefix: configured path prefix for the eXist instance
+    :param root_collection: a path to a collection which will be used as root for all document paths
     :param parser: an lxml etree.XMLParser instance to parse query results
     """
 
     def __init__(
-            self,
-            host: str = DEFAULT_HOST,
-            port: int = DEFAULT_PORT,
-            user: str = DEFAULT_USER,
-            password: str = DEFAULT_PASSWORD,
-            prefix: str = "exist",
-            parser: etree.XMLParser = DEFAULT_PARSER
+        self,
+        host: str = DEFAULT_HOST,
+        port: int = DEFAULT_PORT,
+        user: str = DEFAULT_USER,
+        password: str = DEFAULT_PASSWORD,
+        prefix: str = "exist",
+        root_collection: str = "/",
+        parser: etree.XMLParser = DEFAULT_PARSER,
     ):
-        self._root_collection = "/"
-        self.host = host
-        self.port = port
-        self.user = user
-        self.password = password
-        self.prefix = prefix
+        self.__connection_props = ConnectionProps(
+            root_collection=root_collection,
+            host=host,
+            port=port,
+            user=user,
+            password=password,
+            prefix=prefix,
+        )
+        self._base_url = (
+            f"http://{self.__connection_props.user}"
```

Comment:
you could still use the arguments here since they didn't change. slight perfomance imrovement, and better to read.

Reply:
true. i'll change this